crates/cli/src/native_query/pipeline/mod.rs

```diff
@@ -462,7 +462,7 @@ mod tests {
                         Some(TypeConstraint::ElementOf(Box::new(
                             TypeConstraint::FieldOf {
                                 target_type: Box::new(TypeConstraint::Variable(input_doc_variable)),
-                                path: nonempty!["words".into()],
+                                path: NonEmpty::singleton("words".into()),
```

Comment:
I had to change a bunch of lines from `nonempty![x]` to `NonEmpty::singleton(x)` because nonempty v0.11.0 has a bug in its `nonempty!` macro when applied to a single value. In the places we've been using it we've been using v0.10.0; I could have downgraded back to that version, but v0.11.0 has an `.as_ref()` method that was useful for these changes.I could have used v0.11.0 in the mongo-agent-common and ndc-query-plan crates, and left the others on v0.10.0, but I'd rather use a single dependency version everywhere.